### PR TITLE
Add line_discipline kwarg to job_monitor function

### LINKED ISSUE #5690: more control over job_monitor output
### What is the expected enhancement?
Allow the user to control the new line characters that are emitted by job_monitor output.
I found myself needing to change this to allow me to use qiskit-terra's tools on my non jupyter notebook environment.
I'm already working on a PR for this, which will appear linked here very soon.

### PR DESCRIPTION
**Pull request by omarcostahamido**
### Summary
add line_discipline kwarg to `job_monitor()`

### Details and comments
This PR fixes #5690 by adding an extra kwarg named line_discipline.
It defaults to what was being used already `\r`, so nothing will change.
It will only allow some more control over the job_monitor output.

Fixes #5690

### CODE REVIEW

**Comment by omarcostahamido** [comment]
PR ready for review.
Thanks to @jwoehr !

File: releasenotes/notes/add-line_discipline-to-job_monitor-dc9b66a9d819dbfa.yaml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+---
+features:
+  - |
+    New line_discipline kwarg for job_monitor(). User can now change the new line characters that are emitted by job_monitor output. The line_discipline kwarg defaults to '\r', which is what was in use before. Refer to `#5690 <https://github.com/Qiskit/qiskit-terra/issues/5690>` for more details.
```
Comment (jyu00) [inline releasenotes/notes/add-line_discipline-to-job_monitor-dc9b66a9d819dbfa.yaml:4]:
```suggestion
    New line_discipline kwarg for job_monitor(). User can now change the cartridge return characters in the ``job_monitor`` output. The line_discipline kwarg defaults to '\r', which is what was in use before. Refer to `#5690 <https://github.com/Qiskit/qiskit-terra/issues/5690>` for more details.
```

Comment (omarcostahamido) [reply releasenotes/notes/add-line_discipline-to-job_monitor-dc9b66a9d819dbfa.yaml:4]: thank you

**Comment by omarcostahamido** [comment]
@jyu00 ready

**Comment by jyu00** [comment]
@omarcostahamido Can you fix the lint/doc issues reported by the CI? Thanks.

**Comment by omarcostahamido** [comment]
@jyu00 updated the branch and it now says 
```
Code owner review required
Waiting on code owner review from Qiskit/terra-core
Qiskit/terra-core was requested for review as a code owner
```

**Comment by omarcostahamido** [comment]
ok @jyu00 I've made several changes it asked me to. This last one I can't seem to be able to understand. Can you help?

**Comment by omarcostahamido** [comment]
@jyu00 all checks have passed.

**Comment by jwoehr** [comment]
@jyu00 ..  @omarcostahamido and I have noticed that there are actually 2 similar-but-different versions of `job_monitor`.

1. [`qiskit-terra/qiskit/tools/monitor/job_monitor.py`]( https://qiskit.org/documentation/stubs/qiskit.tools.job_monitor.html#qiskit.tools.job_monitor)<= the subject of this PR
2. [`qiskit-ibmq-provider/qiskit/providers/ibmq/job/job_monitor.py`](https://qiskit.org/documentation/stubs/qiskit.providers.ibmq.job.job_monitor.html#qiskit.providers.ibmq.job.job_monitor) <= cut-and-paste reuse of much the same code
If you are inclined to accept this PR, we probably should issue a similar PR for qiskit-ibmq-provider.

**Comment by jyu00** [comment]
@jwoehr Yes there are 2 `job_monitor`. It started out in `qiskit-terra` but was copied to `qiskit-ibmq-provider` because it only handles IBM Quantum jobs. The copy in `qiskit-terra` was kept with the plan to make it generic for all providers. And yes please submit a similar PR in `qiskit-ibmq-provider`.

**Review by jyu00 — DISMISSED** [review]

**Comment by jwoehr** [comment]
I see in the changes that "carriage return" is misspelled "cartridge return".

> User can now change the cartridge return characters in the ``job_monitor`` output.

**Comment by omarcostahamido** [comment]
@jwoehr good catch. thank you!
fixed.

@jyu00 please approve again (once travis is done).

**Comment by jyu00** [comment]
@1ucian0 I think this needs to be approved by someone in the `terra-core` team to be merged.

File: releasenotes/notes/add-line_discipline-to-job_monitor-dc9b66a9d819dbfa.yaml [hunk 2]
```diff
@@ -0,0 +1,4 @@
+---
+features:
+  - |
+    New line_discipline kwarg for job_monitor(). User can now change the carriage return characters in the ``job_monitor`` output. The line_discipline kwarg defaults to '\r', which is what was in use before. Refer to `#5690 <https://github.com/Qiskit/qiskit-terra/issues/5690>` for more details.
```
Comment (mtreinish) [inline releasenotes/notes/add-line_discipline-to-job_monitor-dc9b66a9d819dbfa.yaml:4]:
```suggestion
    A new kwarg, ``line_discipline``, has been added to the :func:`~qiskit.tools.job_monitor`
    function. This kwarg enables changing the carriage return characters used in the
    ``job_monitor`` output. The ``line_discipline`` kwarg defaults to ``'\r'``, which is what was
     in use before. Refer to `#5690 <https://github.com/Qiskit/qiskit-terra/issues/5690>`__ for more details.
```

**Review by mtreinish — COMMENTED** [review]
LGTM, just one comment on the release note

**Review by mtreinish — DISMISSED** [review]

**Review by mtreinish — APPROVED** [review]

**Comment by omarcostahamido** [comment]
thank you @mtreinish .
Hopefully this will finally be merged 🤞

**Comment by omarcostahamido** [comment]
👇  ?

**Comment by omarcostahamido** [comment]
omg finally.
Thanks @mtreinish

**Pull request merged**